osu.Game.Rulesets.Mania/Difficulty/Skills/Strain.cs

```diff
@@ -4,98 +4,281 @@
 #nullable disable
 
 using System;
-using osu.Framework.Utils;
 using osu.Game.Rulesets.Difficulty.Preprocessing;
 using osu.Game.Rulesets.Difficulty.Skills;
 using osu.Game.Rulesets.Mania.Difficulty.Preprocessing;
 using osu.Game.Rulesets.Mods;
 
+/* Rules of Strain Calculation
+ * 1) The current strain value can only depend on past notes
+ * 2) Notes on the same offset may see different strain values, as column order sorting is not guaranteed.
+ * 3) 1st hit object is OMITTED as it acts as a reference for the following note.
+ *
+ * E.g. A:(100ms @ Col 1), B:(200ms @ Col 1), C:(200ms @ Col 2)
+ * The following sequences are possible: B C, C B.
+ * A is omitted because of rule 3.
+ */
+
+/* Previous Note States
+ * When calculating strain by its history, there's these possible states.
+ *
+ * The column describes where the previous note's end time is.
+ * E.g. E3 states that the previous note's end time is on the body (of the current note)
+ *
+ * Impossible states are marked with X, as their head is AFTER our current offset.
+ *
+ * Legend
+ * ┌──────────────┐
+ * │ (State Name) │
+ * │ Prev Note    │
+ * │ This Note    │
+ * └──────────────┘
+ *                 ┌─────────────┬─────────────┬─────────────┬─────────────┬──────────────┐
+ *                 │ Before Head │ On Head     │ On Body     │  On Tail    │ After Tail   │
+ *                 │ (1)         │ (2)         │ (3)         │  (4)        │ (5)          │
+ * ┌───────────────┼─────────────┼─────────────┼─────────────┼─────────────┼──────────────┤
+ * │ Note      (A) │ (A1)        │ (A2)        │             │             │              │
+ * │ Before        │ O           │      O      │      X      │      X      │      X       │
+ * │ Note          │      O      │      O      │             │             │              │
+ * ├───────────────┼─────────────┼─────────────┼─────────────┼─────────────┼──────────────┤
+ * │ Long Note (B) │ (B1)        │ (B2)        │             │             │ (B5)         │
+ * │ Before        │ [==]        │ [====]      │      X      │      X      │ [==========] │
+ * │ Note          │      O      │      O      │             │             │      O       │
+ * ├───────────────┼─────────────┼─────────────┼─────────────┼─────────────┼──────────────┤
+ * │ Long Note (C) │             │             │             │             │ (C5)         │
+ * │ Before        │      X      │      x      │      X      │      X      │      [===]   │
+ * │ Note          │             │             │             │             │      O       │
```

Comment:
Why B and C are separate rows, despite involving the same notes, is also not explained very well. The same goes for E and F.

Reply:
That's true, I'll see how to annotate it better